sklearn/utils/estimator_checks.py

```diff
@@ -355,10 +358,29 @@ def _is_32bit():
     return struct.calcsize('P') * 8 == 32
 
 
+def gram_matrix_if_pairwise(X, estimator, kernel=linear_kernel):
+
+    if len(X.shape) == 1:
+        X = X.reshape(-1, 1)
+
+    if is_pairwise_metric(estimator):
+        return pairwise_distances(X, metric='mahalanobis')
+    if is_pairwise(estimator):
+        return kernel(X, X)
```

Comment:
FWIW, we do have a `pairwise_kernels` function with configurable distance, comparable to `pairwise_distances`

Reply:
Gotcha. It looks like `pairwise_kernels` only contains similarity metrics though? For KNeighborsRegressor and the like we'd need a distance metric.

In the end it seems like a toss up between 3 options:
- `pairwise_distances` with mahalanobis distance. Makes the most sense but causes the weird Travis failure
- a quick hand written metric, normalizing each dimension and using euclidean distance. Works and doesn't cause any errors, but isn't the most elegant solution. 
- `pairwise_kernel`: We could perhaps use the inverse of the result, since it measures similarity. This will violate the `d(x, y) == 0 if x == y` property of a metric though